### PR TITLE
ASM-3855 Re-factoring iom_8x4 SAN switch puppet module using puppet a…

### PR DESCRIPTION
**Pull request by shariqiqbal1**
…pply

### CODE REVIEW

**Comment by gavin-scott** [comment]
Looks ok to me. @asaenz can you review also?

File: lib/puppet_x/iom_8x4/model/generic_value.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          end
+        end
+
+        def define_value_method(methods)
+          methods.each do |meth|
+            define_method(meth) do |*args, &block|
+              # return the current value if we are called like an accessor
+              return instance_variable_get("@#{meth}".to_sym) if args.empty? && block.nil?
+              # set the new value if there is any
+              instance_variable_set("@#{meth}".to_sym, (block.nil? ? args.first : block))
```
Comment (asaenz) [inline lib/puppet_x/iom_8x4/model/generic_value.rb]: @shariqiqbal1 I'm pretty sure there's some end statements missing here from this method.

File: lib/puppet/util/iom_8x4.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
-require 'puppet/util/network_device'
-
-module Puppet::Util::NetworkDevice::Iom_8x4
+module Puppet
+	module util
+		module Iom_8x4
```
Comment (asaenz) [inline lib/puppet/util/iom_8x4.rb]: @shariqiqbal1 I don't think this module is being used.  Could probably remove, and retest just to be sure.

File: lib/puppet_x/iom_8x4/ssh.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+module PuppetX
+	module Iom8x4
+		class Ssh < Puppet::Util::NetworkDevice::Transport::Ssh
+      #These switches require a carriage return as well, instead of just new line.  So we override Puppet's method to add \r
+      def send(line, noop=false)
+        Puppet.debug "SSH send: #{line}" if Puppet[:Debug]
+        @channel.send_data(line + "\n\r") unless noop
+      end
+
+			def sendwithoutnewline(line, noop = false)
```
Comment (asaenz) [inline lib/puppet_x/iom_8x4/ssh.rb]: @shariqiqbal1 This spacing looks weird.  Check to make sure the tabs are using spaces, and they are consistent (2 spaces is the recommended for Ruby).

**Comment by asaenz** [comment]
@gavin-scott  @shariqiqbal1 Made some changes for my code comments, and squashed down the extraneous commits.

This looks good to go now.  :+1:

**Pull request merged**